core/blockchain/impl/block_tree_impl.cpp

```diff
@@ -55,16 +56,94 @@ namespace kagome::blockchain {
       std::shared_ptr<storage::changes_trie::ChangesTracker> changes_tracker,
       std::shared_ptr<primitives::BabeConfiguration> babe_configuration,
       std::shared_ptr<consensus::BabeUtil> babe_util) {
-    // create meta structures from the retrieved header
-    OUTCOME_TRY(hash, header_repo->getHashById(last_finalized_block));
-    OUTCOME_TRY(number, header_repo->getNumberById(last_finalized_block));
+    BOOST_ASSERT(storage != nullptr);
```

Comment:
I prefer
```suggestion
    BOOST_ASSERT(storage);
```
ofc, that is a matter of taste